src/websocket-connection.js

```diff
@@ -116,14 +116,20 @@ WebsocketConnection.prototype.close = function() {
     return Promise.resolve();
   }
   return new Promise(function(resolve) {
+    this._close();
+    resolve();
+  }.bind(this));
+};
+
+WebsocketConnection.prototype._close = function() {
+  if (!this.isClosed()) {
     if (this._webSocket.terminate) {
       this._webSocket.terminate();
     } else {
       this._webSocket.close();
     }
-    this.emit('close');
-    resolve();
-  }.bind(this));
+  }
+  this.emit('close');
```

Comment:
ok, but i meant that we should probably not emit the "close" event if the connection is already closed.